utils/aws_sigv4.cc

```diff
@@ -98,10 +101,12 @@ std::string get_signature(std::string_view access_key_id, std::string_view secre
 
     std::string payload_hash = apply_sha256(body_content);
     std::string canonical_request = fmt::format("{}\n{}\n{}\n{}\n{}\n{}", method, canonical_uri, query_string, canonical_headers.str(), signed_headers_str, payload_hash);
+    alog.debug("Canonical request: [{}]", canonical_request);
```

Comment:
Yeah, it's going to be one-time logger. Only useful if extending the signing code. There can be two options to ... fix that. First is to patch get_signature() to log into caller's logger, but that's, again, extra tiny perf penalty. Second is to wrap it with `#ifdef DEVEL` to keep it only for dev-builds. Or maybe seastar logger might get its `logger::log_if_not_release()` helper. I don't have good ideas either :(

Reply:
I am not sure it has significant performance penalty.

AFAICT, logger uses a lazy approach, where the arguments are taken by reference and will only invoke the ostream formatter if the tagged level is <= than the configured logging level.

Kefu even optimized the formmater for basic_sstring in  https://github.com/scylladb/seastar/commit/bef98ab006e4a10d4f2811f94ec611b368ec233b, and note it's only relevant if the message is going to be dumped in the log. Otherwise, formatter is not invoked.

Reply:
There is the (I admitted it's not big) penalty of a if(), and more code.  I don't mind it if you think anyone will ever use this logger. I just don't think that everything you wanted to print during your debugger will always be useful for others. But maybe. As I said, I don't feel strongly about it.